includes/class-wc-payments-account.php

```diff
@@ -643,4 +646,53 @@ private function settings_changed( $changes = [] ) {
 		$diff = array_diff_assoc( $changes, $account );
 		return ! empty( $diff );
 	}
+
+	/**
+	 * Returns an array containing the names of all the WCPay related notes that have be actioned.
+	 *
+	 * @return array
+	 */
+	private function get_actioned_notes() {
+		$wcpay_promo_notes = [];
+
+		try {
+			/**
+			 * Data Store for admin notes
+			 *
+			 * @var DataStore $data_store
+			 */
+			$data_store = WC_Data_Store::load( 'admin-note' );
+		} catch ( Exception $e ) {
+			// Don't stop the on-boarding process if something goes wrong here. Log the error and return the empty array
+			// of actioned notes.
+			Logger::error( $e );
+			return $wcpay_promo_notes;
+		}
+
+		// Get the last 50 actioned notes (we don't want to get all the notes, or rely on the site's per_page setting).
+		$notes = $data_store->get_notes(
```

Comment:
Can we move this into try-clause, so we get something like `$notes = WC_Data_Store::load( 'admin-note' )->get_notes(...)`. It'll greatly simplify the method as we can omit the if below and just force-cast notes to array in foreach?

Reply:
> @jrodger: I'd go for using `woocommerce_note_where_clauses` solution:
> 
> * then limit can be reduced (we limit exactly promo notes)
> * fewer data will be transferred between PHP and DB (resources-friendly)

It's certainly a better solution for those reasons. I was a bit worried about messing around with the SQL query as a string, but I think that's outweighed by your points there.

Reply:
If we go with the where clause approach then we won't need to foreach loop. However, we'd still want to cast the array before it was returned.

Were you thinking something like `return (array)$wcpay_promo_notes;`?

Reply:
Indeed. I was thinking about `foreach ( (array) $notes as ...)`